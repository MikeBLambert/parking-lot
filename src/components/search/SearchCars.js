import React, { Fragment } from 'react';


export default function SearchCars({ updateSearchTerm, searchTerm, car }) {

  const carDetails = car.map(car => {
    return (
      <tr key={car.plate}>
        <th>{car.make}</th>
        <th>{car.model}</th>
        <th>{car.plate}</th>
      </tr>
    );
  });

  return (
    <Fragment>
      <label>Search Cars by Plate</label>
      <input type="text" value={searchTerm} onChange={updateSearchTerm}/>
      {car &&
        <table width="100%">
          <tbody>
            <tr>
              <th>Make</th>
              <th>Model</th>
              <th>Plate</th>
            </tr>
            {car && carDetails}
          </tbody>
        </table>
      }



    </Fragment>
  );
}
