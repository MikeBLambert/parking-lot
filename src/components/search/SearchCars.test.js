import React from 'react';
import { shallow } from 'enzyme';
import SearchCars from './SearchCars';

describe('SearchCars component', () => {
  it('renders a component', () => {
    const car = [{ make: 'Toyota', model: 'Corolla', plate: 'ABC123' }];
    const wrapper = shallow(<SearchCars car={car}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
