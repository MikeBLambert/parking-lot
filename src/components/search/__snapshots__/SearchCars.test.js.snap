// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCars component renders a component 1`] = `
<Fragment>
  <label>
    Search Cars by Plate
  </label>
  <input
    type="text"
  />
  <table
    width="100%"
  >
    <tbody>
      <tr>
        <th>
          Make
        </th>
        <th>
          Model
        </th>
        <th>
          Plate
        </th>
      </tr>
      <tr>
        <th>
          Toyota
        </th>
        <th>
          Corolla
        </th>
        <th>
          ABC123
        </th>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
